src/background/errors.js

```diff
@@ -12,6 +12,8 @@ export class BlockedSiteError extends ExtensionError {}
 
 export class AlreadyInjectedError extends ExtensionError {}
 
+export class BadgeRequestError extends Error {}
```

Comment:
The name here suggests that this could refer to _any_ error that happened when fetching the badge count, but from the implementation it looks like this is only used if the badge request is canceled? If so I would suggest to rename this to reflect that - the exception could either specifically reflect the fact that the badge request was canceled (maybe `BadgeRequestCanceledError`?) or perhaps a more generic exception to reflect that an operation was explicitly canceled (`CanceledError` ?).

Reply:
I think the combination of `Cancel` and `Error` is not clear, it seems to indicate an error raised _during_ the cancelation. I would prefer `RequestError` as a more generic error.